src/libtls/tls_crypto.c

```diff
@@ -2426,3 +2426,51 @@ tls_named_group_t tls_ec_group_to_curve(diffie_hellman_group_t group)
 	}
 	return 0;
 }
+
+/**
+ * See header.
+ */
+array_t *tls_create_common_key_type(chunk_t hashsig, tls_version_t min_version,
+									tls_version_t max_version)
+{
+	bio_reader_t *reader;
+	array_t *types;
+	uint16_t current;
+
+	types = array_create(sizeof(key_type_t), 1);
+	reader = bio_reader_create(hashsig);
+	while (reader->remaining(reader) &&
+		   reader->read_uint16(reader, &current))
+	{
+		int i;
+
+		for (i = 0; i < countof(schemes); i++)
```

Comment:
I extracted this functionality from `tls_server.c` to `tls_crypto.c` because of the existing `tls_signature_scheme_t`-to-`signature_params_t` mapping in `schemes`.